OpenSim/Simulation/SimbodyEngine/Test/testJoints.cpp

```diff
@@ -1946,11 +1946,12 @@ void testAutomaticJointReversal()
     //thigh.addGeometry(Cylinder(0.035, 0.4));
     thigh->upd_geometry(0).setColor(SimTK::Vec3(0, 0, 1));   // BLUE
     
-    //TODO fix Bug: ModleComponent::addGeometry assumes ownership but does not
-    // take a pointer to heap allocated Geometry
-    shank->addGeometry(*new Cylinder(0.02, 0.243800));
+    //ModelComponent::addGeometry makes a copy of the passed in Geometry
+    Cylinder cyl(0.02, 0.243800);
+    shank->addGeometry(cyl);
```

Comment:
The old code snippet was leaking because it was assuming the ModelComponent was taking ownership of the passed in Geometry, the new code doesn't leak since the geometry is allocated on the stack and freed when out of scope, and the copy made downstream is handled by the model (properly I assume)..